src/utils/httpRequestParser.ts

```diff
@@ -100,7 +114,16 @@ export class HttpRequestParser implements IRequestParser {
         // parse body
         const contentTypeHeader = getHeader(headers, 'content-type') || getHeader(this._restClientSettings.defaultHeaders, 'content-type');
         body = HttpRequestParser.parseRequestBody(bodyLines, requestAbsoluteFilePath, contentTypeHeader);
-        if (this._restClientSettings.formParamEncodingStrategy !== FormParamEncodingStrategy.Never && body && typeof body === 'string' && MimeUtility.isFormUrlEncoded(contentTypeHeader)) {
+        if (isGraphQlRequest) {
+            variables = HttpRequestParser.parseRequestBody(variableLines, requestAbsoluteFilePath, contentTypeHeader);
+
+            let graphQlPayload = {
+                query: body,
+                variables: JSON.parse(variables.toString())
+            };
+            body = JSON.stringify(graphQlPayload, null, 2);
+            bodyLines = body.split(EOL);

```

Comment:
Use the `bodyLines` parsed in line 86, and in this case, no need to prettify the graphQlPayload in line 124?

Reply:
`bodyLines` here is a bit different. since your actually converting `graphQlPayload`, which is different than what's on line 86

Reply:
I know your point. However, the `bodyLines` variable is only used as a parameter in `HttpRequest` constructor which stands for the `rawBody`. It is used to record the original body lines in request

Reply:
no need to `prettify`, just came in handy when debugging GraphQL requests.